File: commons/identity/src/main/java/net/blueshell/identity/IdentityProvider.java
package net.blueshell.identity;

import net.blueshell.enums.Role;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;

public abstract class IdentityProvider {
    protected Identity getPrincipal() {
        Object obj = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        if (obj instanceof Identity) {
            return (Identity) obj;
        }
        return null;
    }

    protected boolean hasAuthority(Role role) {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        if (authentication == null) {
            return false;
        }
        return authentication.getAuthorities().stream()
                .anyMatch(a -> a.getAuthority().equals(role.toString()));
    }
}

File: api/src/main/kotlin/net/blueshell/api/auth/security/IdentityProvider.kt
package net.blueshell.api.auth.security

import net.blueshell.api.shared.enums.Role
import net.blueshell.api.user.model.User
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.context.SecurityContextHolder

abstract class IdentityProvider {
    protected val principal: User?
        get() {
            val obj = SecurityContextHolder.getContext().authentication.principal

            if (obj is User) {
                return obj
            }
            return null
        }

    protected fun hasAuthority(role: Role): Boolean {
        val authentication = SecurityContextHolder.getContext().authentication
        return authentication != null && authentication.authorities.stream()
            .anyMatch { a: GrantedAuthority? -> a!!.authority == role.toString() }
    }
}